mongodbatlas/resource_mongodbatlas_private_endpoint_regional_mode.go

```diff
@@ -0,0 +1,183 @@
+		ReadContext:   resourceMongoDBAtlasPrivateEndpointRegionalModeRead,
+		UpdateContext: resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate,
+		DeleteContext: resourceMongoDBAtlasPrivateEndpointRegionalModeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasPrivateEndpointRegionalModeImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId(d.Get("project_id").(string))
+	err := resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate(ctx, d, meta)
+
+	if err != nil {
+		return err
+	}
+
+	return resourceMongoDBAtlasPrivateEndpointRegionalModeRead(ctx, d, meta)
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Id()
+
+	setting, resp, err := conn.PrivateEndpoints.GetRegionalizedPrivateEndpointSetting(context.Background(), projectID)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf(errorPrivateEndpointRegionalModeRead, projectID, err))
+	}
+
+	if err := d.Set("enabled", setting.Enabled); err != nil {
+		return diag.FromErr(fmt.Errorf(errorPrivateEndpointRegionalModeSetting, "enabled", projectID, err))
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Id()
+	enabled := d.Get("enabled").(bool)
+
+	_, resp, err := conn.PrivateEndpoints.UpdateRegionalizedPrivateEndpointSetting(ctx, projectID, enabled)
+	if err != nil {
+		if resp != nil && resp.Response.StatusCode == 404 {
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf(errorPrivateLinkEndpointsDelete, projectID, err))
+	}
+
+	log.Println("[INFO] Waiting for MongoDB Private Endpoints Connection to be destroyed")
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"PENDING", "REPEATING"},
+		Target:     []string{"APPLIED"},
+		Refresh:    resourcePrivateEndpointRegionalModeRefreshFunc(ctx, conn, projectID),
+		Timeout:    1 * time.Hour,
+		MinTimeout: 5 * time.Second,
+		Delay:      3 * time.Second,
+	}
+	// Wait, catching any errors
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(errorPrivateLinkEndpointsDelete, projectID, err))
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasPrivateEndpointRegionalModeDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId("")
+
+	return nil
+}
```

Comment:
[q] Shouldn’t we disable regional mode here?

Reply:
If you or someone else knows that the behavior for delete should be what your question suggests, let me know and I can make that happen ASAP.

Will want to update testing accordingly if that's the case as well

Reply:
I don't have a strong opinion here, my comment was more like a head-up. I will leave the comment unresolved to see what melissa thinks about it. Thanks for the clarification!

Reply:
These are tough as when we've hit them in the past different users will have different assumptions about what delete will mean if it's not a specific resource that can be removed.  I tend to take the position that if you change the tf config you are asking for something to be deleted, if possible, or disabled, if that's the only option.  If a user wants something removed from state then they should do that directly and remove the associated config.  I also lean on your experience @evertsd and @martinstibbe on what other companies have done in similar situations (I've seen both out there) but I think for this if they delete we should disable it.

Reply:
Sounds good, I can make that happen, will request re-review once its done

Reply:
Ok, delete now updates the endpoint to be `enabled: false`, and there is an assertion in the test to ensure it'c changed from true to false on delete.

Reply:
@andreaangiolillo fyi